hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -227,11 +227,7 @@ public String getBootstrapIndexByFileIdFolderNameFolderPath() {
    */
   public String getArchivePath() {
     String archiveFolder = tableConfig.getArchivelogFolder();
-    if (archiveFolder.equals(HoodieTableConfig.ARCHIVELOG_FOLDER.defaultValue())) {
```

Comment:
@danny0405 this is more of a debt due to how we did this at uber originally. Since HDFS supports append, we started writing out these archived logs into a single log file (with rollover) under `.hoodie` itself. But on cloud storage, there is no append and it leads to explosion of small files. So we moved it to the `.archived` subfolder. My take would be leave this alone here and override it at the higher layers like cli, spark, flink for backwards compat reasons. No other reasons. 

cc @n3nash , do you have a preference ?